perllib/Open311/GetServiceRequests.pm

```diff
@@ -141,49 +134,64 @@ sub create_problems {
             );
         };
         $created_time = $updated_time if $@;
+        my $created = DateTime::Format::W3CDTF->format_datetime(
+            $created_time->clone->set_time_zone('UTC')
+        );
+
+        # Updated time must not be before created time, check and adjust as necessary.
+        # (This has happened with some fetched reports, oddly.)
+        $updated = $created if $updated lt $created;
 
         my $problems;
         my $criteria = {
             external_id => $request_id,
         };
-        $problems = $self->schema->resultset('Problem')->to_body($body)->search( $criteria );
+
+        # Skip if this problem already exists (e.g. it may have originated from FMS and is being mirrored back!)
+        next if $self->schema->resultset('Problem')->to_body($body)->search( $criteria )->count;
+
+        if ($args->{start_date} && $args->{end_date} && ($created lt $args->{start_date} || $created gt $args->{end_date}) ) {
```

Comment:
The dates given to this script are used, if I understand it right, to get all requests that have status changes (by LoggedTime in Confirm integration) between them. So should this be using `$updated` as it was previously, not `$created`? As I can see old reports being returned from this.

Reply:
Good point!